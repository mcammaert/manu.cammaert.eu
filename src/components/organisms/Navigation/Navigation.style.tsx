import styled from 'styled-components';
import { rem, rgba, modularScale } from 'polished';
import { Link as LinkComponent } from 'react-router-dom';
import { animated } from 'react-spring';

import { MenuIcon as StyledMenuIcon } from 'components/atoms/MenuIcon';

export const NavigationContainer = styled.div``;

export const Navigation = styled(animated.div)`
  background-color: ${({ theme }) => rgba(theme.colors.secondaryInvert, 0.85)};
  position: fixed;
  z-index: 999;
  transition: backdrop-filter 100ms;
  backdrop-filter: blur(2px);
`;

export const ToggleButton = styled.button`
  position: fixed;
  -webkit-transform: translateZ(0);
  left: ${({ theme }) => rem(theme.margin)};
  top: ${({ theme }) => rem(theme.margin)};
  width: 44px;
  height: 44px;
  z-index: 1000;
  border: none;
  background: ${({ theme }) => theme.colors.primaryInvert};
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0;
  cursor: pointer;
  transition: all 400ms;
  &:hover,
  &:focus {
    background: ${({ theme }) => rgba(theme.colors.primary, 0.035)};
  }
`;

export const Link = styled(LinkComponent)`
  text-decoration: none;
  display: block;
  padding: ${modularScale(0)};
  font-size: ${modularScale(0)};
  transition: all 200ms;
  font-weight: 400;
  color: ${({ theme }) => theme.colors.primary};
  background: ${({ theme }) => theme.colors.primaryInvert};
  &:hover {
    box-shadow: 0 1px 18px ${({ theme }) => rgba(theme.colors.primary, 0.08)};
  }
  &:active {
    color: ${({ theme }) => theme.colors.primaryInvert};
    background: ${({ theme }) => theme.colors.primary};
  }
`;

export const Menu = styled.ul`
  list-style: none;
  padding: ${modularScale(0)};
  padding-top: ${modularScale(6)};
`;

export const MenuItem = styled(animated.li)`
  list-style: none;
  margin-bottom: ${modularScale(-1)};
`;

export const MenuIcon = styled(StyledMenuIcon)``;

export default { NavigationContainer, Navigation, ToggleButton, Link, Menu, MenuItem, MenuIcon };
