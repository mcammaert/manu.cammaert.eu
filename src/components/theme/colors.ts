export const colors = {
  primary90: '#0c2d48',
  primary80: '#0b395d',
  primary70: '#0b4573',
  primary60: '#0d5189',
  primary50: '#145da0',
  primary40: '#367bb1',
  primary30: '#5c99c1',
  primary20: '#85b7d0',
  primary10: '#b1d4e0',

  secondary90: '#ca3f36',
  secondary80: '#d35239',
  secondary70: '#db633e',
  secondary60: '#e37443',
  secondary50: '#eb844a',
  secondary40: '#f79b77',
  secondary30: '#fdb5a1',
  secondary20: '#fdd0c8',
  secondary10: '#f7eceb',

  accent1: '#d45e95',
  accent2: '#b290d4',
  accent3: '#97b8e6',

  // 10 shades of grey
  black: '#1E2025',
  charcoal: '#262830',
  shadow: '#2E313A',
  lead: '#585965',
  steel: '#787C8A',
  gray: '#989DAC',
  smoke: '#BEC2C9',
  fog: '#EDEEF0',
  white: '#F6F7F9',
  snow: '#FCFCFC',

  // slate
  slateLight: '#3279AD',
  slate: '#4C6E89',
  slateDark: '#455C77',

  // teal
  teal: '#3D829C',
  tealDark: '#36768B',

  // laguna
  lagunaLight: '#4FB8F3',
  laguna: '#57A7E3',
  lagunaDark: '#3680DC',

  // warm colors
  red: '#F60C5B',
  orange: '#F86C5C',
  sunflower: '#EDC172',
  lemon: '#FFE362',

  // purples
  grape: '#374278',
  plum: '#725FA4',
  lavender: '#936FBA',
  purple: '#C163C6',

  // greens
  forest: '#50C04E',
  lawn: '#86DE6A',
  aquamarine: '#14DAB5',
  emerald: '#0DCACE',
  turquoise: '#57D5D6',

  // browns
  chocolate: '#3C322D',
  barnwood: '#5E5E5F',
};

export default colors;
