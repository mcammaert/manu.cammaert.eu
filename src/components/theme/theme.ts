import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  body {
    background-color: #ccc;
  }
`;

const theme = {
  main: 'mediumseagreen',
};

export default theme;
