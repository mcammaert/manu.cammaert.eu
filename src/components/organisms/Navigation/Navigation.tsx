import React, { useState, useRef } from 'react';

import { useLockBodyScroll } from 'react-use';
import { interpolate, useTransition, animated, useTrail, ReactSpringHook, useChain, config } from 'react-spring';

import { ScreenReaderOnly } from 'components/atoms/ScreenReaderOnly';
import { NavigationProps } from './Navigation.types';

import S from './Navigation.style';

const links = [
  {
    title: 'Home',
    uri: '/',
  },
  {
    title: 'Curriculum Vitae',
    uri: '/cv',
  },
  {
    title: 'Contact',
    uri: '/contact',
  },
];

const Navigation: React.FC<NavigationProps> = () => {
  const [showNavigation, setShowNavigation] = useState(false);

  const onToggleNavigationHandler = () => {
    setShowNavigation(!showNavigation);
  };

  useLockBodyScroll(showNavigation);

  const basicRef = {
    start: () => {},
    stop: () => {},
  };

  const transRef = useRef<ReactSpringHook>(basicRef);
  const transitions = useTransition(showNavigation, null, {
    from: { position: 'absolute', opacity: 0, zIndex: 999, left: 0, top: 0, width: '100vw', height: '100vh' },
    enter: { opacity: 1 },
    leave: { opacity: 0 },
    ref: transRef,
    config: config.stiff,
  });

  const trailRef = useRef<ReactSpringHook>(basicRef);
  const trail = useTrail(links.length, {
    opacity: showNavigation ? 1 : 0,
    x: showNavigation ? 0 : 20,
    ref: trailRef,
    trail: 100 / links.length,
    config: config.stiff,
  });

  useChain(showNavigation ? [transRef, trailRef] : [trailRef, transRef], [0, showNavigation ? 0.1 : 0.9]);

  const clickLinkHandler = () => {
    setShowNavigation(false);
  };

  const MenuItem = ({ title, uri }: { title: string; uri: string }) => (
    <S.MenuItem>
      <S.Link to={uri} onClick={clickLinkHandler}>
        {title}
      </S.Link>
    </S.MenuItem>
  );

  return (
    <S.NavigationContainer>
      <S.ToggleButton onClick={onToggleNavigationHandler}>
        <S.MenuIcon checked={showNavigation} />
        <ScreenReaderOnly>{showNavigation ? 'Verberg navigatie' : 'Toon navigatie'}</ScreenReaderOnly>
      </S.ToggleButton>
      <>
        {transitions.map(
          ({ item, props, key }) =>
            item && (
              <animated.div key={key} style={props}>
                <S.Navigation>
                  <S.Menu>
                    {trail.map(({ x, ...rest }, index) => {
                      const { uri, title } = links[index];
                      return (
                        <animated.div
                          key={uri}
                          style={{ ...rest, transform: interpolate([x], (value: number) => `translate3d(0,${value}px,0)`) }}
                          // eslint-disable-next-line react/jsx-props-no-spreading
                          {...rest}>
                          <MenuItem uri={uri} title={title} />
                        </animated.div>
                      );
                    })}
                  </S.Menu>
                </S.Navigation>
              </animated.div>
            ),
        )}
      </>
    </S.NavigationContainer>
  );
};

export default Navigation;
