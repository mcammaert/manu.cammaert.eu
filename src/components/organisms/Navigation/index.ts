import * as S from './Navigation.style';

export { default as Navigation } from './Navigation';
export * from './Navigation.types';
export { S };
