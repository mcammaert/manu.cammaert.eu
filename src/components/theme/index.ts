import { createGlobalStyle } from 'styled-components';
import 'normalize.css';
import { colors } from './colors';

export interface GlobalStyleProps {
  readonly light: boolean;
}

// const backgroundColor = (location: string) => (location === '/' ? colors.oxfordBlue : colors.fuzzyWuzzy);

export const GlobalStyle = createGlobalStyle<GlobalStyleProps>`
  body {
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    font-family: 'Nunito Sans', sans-serif;
    width: 100vw;
    background: ${({ light }) => (light ? colors.white : colors.black)};
    background-attachment: fixed;    
    color: ${({ light }) => (light ? colors.black : colors.snow)};
    transition: all 800ms;
    a {
      color: ${({ theme }) => theme.colors.secondary};
    }
  }
`;

export const lightTheme = {
  margin: 15,
  colors: {
    primary: colors.charcoal,
    primaryInvert: colors.snow,
    secondary: colors.primary60,
    secondaryInvert: colors.white,
    tertiary: colors.secondary80,
    tertiaryInvert: colors.white,
    copy: colors.charcoal,
    copyInvert: colors.white,
  },
};

export const darkTheme = {
  margin: 15,
  colors: {
    primary: colors.snow,
    primaryInvert: colors.black,
    secondary: colors.primary30,
    secondaryInvert: colors.primary90,
    tertiary: colors.secondary30,
    tertiaryInvert: colors.secondary80,
    copy: colors.white,
    copyInvert: colors.charcoal,
  },
};
