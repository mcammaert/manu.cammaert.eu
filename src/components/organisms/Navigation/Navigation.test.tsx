import React from 'react';

import { BrowserRouter } from 'react-router-dom';
import { cleanup } from '@testing-library/react';
import 'jest-styled-components';

import { renderComponent } from 'testing/react-testing-library';
import theme from 'components/theme/light';

import Navigation from './Navigation';

afterEach(cleanup);

describe('Navigation', () => {
  describe('default props', () => {
    it('should match the snapshot', () => {
      const component = (
        <BrowserRouter>
          <Navigation />
        </BrowserRouter>
      );
      const { container } = renderComponent({ theme, component });
      expect(container).toMatchSnapshot();
    });
  });
});
