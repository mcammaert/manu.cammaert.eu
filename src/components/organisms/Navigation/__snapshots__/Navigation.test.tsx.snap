// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation default props should match the snapshot 1`] = `
.c3 {
  position: absolute;
  left: -10000px;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 0;
}

.c6 {
  width: 100vw;
  max-width: 800px;
  padding-left: 0.9375rem;
  padding-right: 0.9375rem;
  margin-top: 0;
  margin-bottom: 0;
}

.c1 {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c2 {
  stroke: #262830;
  stroke-width: 2;
}

.c4 {
  background-color: rgba(244,245,247,0.85);
  position: fixed;
  z-index: 999;
  -webkit-transition: backdrop-filter 100ms;
  transition: backdrop-filter 100ms;
  -webkit-backdrop-filter: blur(2px);
  backdrop-filter: blur(2px);
}

.c0 {
  position: fixed;
  -webkit-transform: translateZ(0);
  left: 0.9375rem;
  top: 0.9375rem;
  width: 44px;
  height: 44px;
  z-index: 1000;
  border: none;
  background: #FCFCFC;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  cursor: pointer;
  -webkit-transition: all 400ms;
  transition: all 400ms;
}

.c0:hover,
.c0:focus {
  background: rgba(38,40,48,0.035);
}

.c7 {
  list-style: none;
  padding: 1em;
  padding-top: 5.610232974924882em;
}

<div>
  <div
    class=""
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="-1"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="-1"
    />
    <div
      data-focus-lock-disabled="disabled"
    >
      <button
        class="c0"
      >
        <svg
          class="c1 "
          height="20"
          width="20"
        >
          <line
            class="c2"
            x1="1"
            x2="19"
            y1="1"
            y2="1"
          />
          <line
            class="c2"
            x1="0.5"
            x2="14"
            y1="10"
            y2="10"
          />
          <line
            class="c2"
            x1="1"
            x2="19"
            y1="19"
            y2="19"
          />
        </svg>
        <div
          class="c3"
        >
          Toon het menu
        </div>
      </button>
      <div
        aria-modal="false"
        class="c4"
        role=""
        style="opacity: 0; top: 0px; right: 0px; bottom: 768px; left: 0px;"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <ul
              class="c7"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="-1"
    />
  </div>
</div>
`;
